### PR TITLE
[master][console] Allow multiple options to be set.

### PR DESCRIPTION
**Pull request by SpacePossum**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| License | MIT |

This PR add the possibility to set multiple options on a formatted string output.
Example:

``` php
$output->writeln('<fg=green;options=bold,underscore>Test</>');
```

Secondly it makes the behavior on invalid values consistent. 

``` php
// current
$output->writeln('<fg=lime;>Test</>'); // throws exception
$output->writeln('<options=italic;>Test</>'); // silent ignore

// new
$output->writeln('<fg=lime;>Test</>'); // throws exception
$output->writeln('<options=italic;>Test</>'); // throws exception
```

All other changes are about making the code more strict or other SCA/CS fixes.

### CODE REVIEW

**Comment by fabpot** [comment]
Anything not related to this new feature should be removed as it would make merging old branches more difficult (and they don't bring any value anyway).

File: src/Symfony/Component/Console/Tests/Formatter/OutputFormatterTest.php [hunk 1]
```diff
@@ -152,6 +153,71 @@ public function testInlineStyle()
...
+            array('<fg=green;options=bold>', "\033[32;1mb\033[39;22m", 'b'),
+            array('<fg=green;options=reverse;>', "\033[32;7m<a>\033[39;27m", '<a>'),
+            array('<fg=green;options=bold,underscore>', "\033[32;1;4mz\033[39;22;24m", 'z'),
+            array('<fg=green;options=bold,underscore,reverse;>', "\033[32;1;4;7md\033[39;22;24;27m", 'd'),
+        );
+    }
+
+    /**
+     * @group legacy
+     * @dataProvider provideInlineStyleTagsWithUnknownOptions
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Tests/Formatter/OutputFormatterTest.php]: missing `@requires function Symfony\Bridge\PhpUnit\ErrorAssert::assertDeprecationsAreTriggered` annotation

Comment (SpacePossum) [inline src/Symfony/Component/Console/Tests/Formatter/OutputFormatterTest.php]: done,
just for my understanding, why is this needed?

Comment (HeahDude) [inline src/Symfony/Component/Console/Tests/Formatter/OutputFormatterTest.php]: @SpacePossum when tests run in max depth (latest versions of the components supported for an older package) this function might not be available from the bridge.

Comment (SpacePossum) [inline src/Symfony/Component/Console/Tests/Formatter/OutputFormatterTest.php]: make sense, thanks for the info @HeahDude !

**Comment by SpacePossum** [comment]
rebased and all comments addressed

**Comment by fabpot** [comment]
Thank you @SpacePossum.
